randovania/games/samus_returns/exporter/patch_data_factory.py

```diff
@@ -42,28 +42,23 @@
 
 def get_item_id_for_item(item: ResourceInfo) -> str:
     assert isinstance(item, ItemResourceInfo)
-    if "item_capacity_id" in item.extra:
+    if "item_export_id" in item.extra:
+        return item.extra["item_export_id"]
+    elif "item_capacity_id" in item.extra:
         return item.extra["item_capacity_id"]
     try:
         return item.extra["item_id"]
     except KeyError as e:
         raise KeyError(f"{item.long_name} has no item ID.") from e
 
 
-def get_export_item_id_for_item(item: ResourceInfo) -> str:
-    assert isinstance(item, ItemResourceInfo)
-    if "item_export_id" in item.extra:
-        return item.extra["item_export_id"]
-    return get_item_id_for_item(item)
```

Comment:
Was there a reason to change this? My intention was to differentiate between the exported id for the patcher + temporary locked ressources and the item id which needs to be tracked if you want to know how many ammo you have.

Reply:
To me, it looks like it does what was done before but without an extra function call. It still produces a correct patcher file which is why I went with it.

Reply:
`get_item_id_for_item` was able to return the `item_capacity_id` even if it had an `item_export_id` e.g. if you want to auto track the amount of usable missiles which is`ITEM_WEAPON_MISSILE_MAX`. In that case you would need the `item_capacity_id` even an `item_export_id` exists.
Now, this function is not usable anymore for it because it will always return the `item_export_id` of such items.
If you want to go with this change and we don't want to track it anyways, you could completly remove the `item_export_id` from the db and place the string of it into the `item_capacity_id`.

Reply:
Is there a benefit to keeping track? I figured it was ok to make this change.

Reply:
Well, uh, the benefit of keeping track is that you keeping track of it xD. I mean, it's the purpose of an auto tracker